keep-ui/app/providers/provider-form.tsx

```diff
@@ -480,7 +480,18 @@ const ProviderForm = ({
         })
         .catch((error) => {
           const updatedFormErrors = error.toString();
-          setFormErrors(updatedFormErrors);
+
+          if (updatedFormErrors.includes("SyntaxError")) {
+            setFormErrors(
+              "Bad response from API: Check the backend logs for more details"
+            );
+          } else if (updatedFormErrors.includes("Failed to fetch")) {
+            setFormErrors(
+              "Failed to connect to API: Check your internet connection"
```

Comment:
Some providers have the API host setting and it could be the problem.
```suggestion
              "Failed to connect to API: Check provider settings and your internet connection"
```